stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift

```diff
@@ -69,34 +69,39 @@ public func _stdlib_thread_barrier_init(
   InitializeConditionVariable(barrier.pointee.cond!)
 #else
   barrier.pointee.mutex = UnsafeMutablePointer.allocate(capacity: 1)
-  if pthread_mutex_init(barrier.pointee.mutex!, nil) != 0 {
-    // FIXME: leaking memory.
-    return -1
-  }
   barrier.pointee.cond = UnsafeMutablePointer.allocate(capacity: 1)
-  if pthread_cond_init(barrier.pointee.cond!, nil) != 0 {
-    // FIXME: leaking memory, leaking a mutex.
-    return -1
+  guard _stdlib_thread_barrier_mutex_and_cond_init(barrier) == 0 else {
+    barrier.pointee.mutex!.deinitialize(count: 1)
+    barrier.pointee.mutex!.deallocate()
+    barrier.pointee.cond!.deinitialize(count: 1)
+    barrier.pointee.cond!.deallocate()
```

Comment:
This has to exit the enclosing scope.

Reply:
Thanks! Rebasing isn't as straightforward as I thought, I'll try building locally this time